lib/net.c

```diff
@@ -1517,19 +1517,23 @@ start_port(uint8_t port_id, uint8_t *pnum_succ_ports,
 	 * are activated on a timer, so this can take some time.
 	 */
 	do {
-		rte_eth_link_get(port_id, &link);
+		ret = rte_eth_link_get(port_id, &link);
+		if (ret < 0) {
+			G_LOG(ERR, "net: querying port %hhu failed with err - %s\n",
+				port_id, rte_strerror(-ret));
+		}
 
 		/* Link is up. */
-		if (link.link_status)
+		if (ret = 0 && link.link_status) {
```

Comment:
Notice that using `=` instead of `==` here is a bug. But instead of replacing `=` with `==`, adding line `RTE_VERIFY(ret == 0);` immediately after `}` of the statement `if (ret < 0) {` is a more robust approach.

Once line `RTE_VERIFY(ret == 0);` is added, this `if` statement doesn't need any change, so remove `ret = 0 && ` and the braces.